### PR TITLE
Addresses mol2 reading bug #944

### PR DESCRIPTION
**Pull request by jchodera**

### CODE REVIEW

**Comment by jchodera** [comment]
Addresses #944

**Comment by jchodera** [comment]
I can't seem to run this on my own `osx` box.  Any idea what this means?
```
mski1776:adp-test choderaj$ python test_mdtraj.py
Traceback (most recent call last):
  File "test_mdtraj.py", line 3, in <module>
    import mdtraj
  File "/Users/choderaj/miniconda/lib/python2.7/site-packages/mdtraj-1.5.0.dev1-py2.7-macosx-10.5-x86_64.egg/mdtraj/__init__.py", line 29, in <module>
    from mdtraj.formats.registry import _FormatRegistry
  File "/Users/choderaj/miniconda/lib/python2.7/site-packages/mdtraj-1.5.0.dev1-py2.7-macosx-10.5-x86_64.egg/mdtraj/formats/__init__.py", line 13, in <module>
    from mdtraj.formats.dtr import DTRTrajectoryFile
ImportError: dlopen(/Users/choderaj/miniconda/lib/python2.7/site-packages/mdtraj-1.5.0.dev1-py2.7-macosx-10.5-x86_64.egg/mdtraj/formats/dtr.so, 2): Symbol not found: __ZNKSt5ctypeIcE13_M_widen_initEv
  Referenced from: /Users/choderaj/miniconda/lib/python2.7/site-packages/mdtraj-1.5.0.dev1-py2.7-macosx-10.5-x86_64.egg/mdtraj/formats/dtr.so
  Expected in: /usr/lib/libstdc++.6.dylib
```

**Comment by rmcgibbo** [comment]
Is this the new apple OS?

**Comment by jchodera** [comment]
10.10.5, so "newish" but not "new"

**Comment by rmcgibbo** [comment]
try blowing away the build directory, `export CC=clang`, and recompiling.

**Comment by rmcgibbo** [comment]
(my guess is that you might be using GNU gcc?)

**Comment by jchodera** [comment]
It looks like the `status` field can be empty in some `mol2` files.  Is there a way to have `read_csv` use a variable number of fields in 
```python
    atoms_frame = pd.read_csv(csv, sep="\s*", engine='python',  header=None,
        names=["serial", "name", "x", "y", "z",
               "atype", "code", "resName", "charge"])
```
?

**Comment by rmcgibbo** [comment]
lol idk.

**Comment by jchodera** [comment]
I still think that a mol2 file parser that does not conform to the mol2 file spec might be more dangerous than not having one at all...

**Comment by jchodera** [comment]
OK, think I fixed it.  Both types of mol2 files work on my machine.

I've added a test for the issue too.

**Comment by jchodera** [comment]
> try blowing away the build directory, export CC=clang, and recompiling.

BTW: Thanks, that worked!

**Comment by mpharrigan** [comment]
Do you really need to read twice?

**Comment by jchodera** [comment]
> Do you really need to read twice?

If I knew what the fuck I was doing with pandas, no.

**Comment by jchodera** [comment]
But I don't, so yes.

**Comment by jchodera** [comment]
This whole thing is a big ball of hacky hacks anyway.

**Comment by jchodera** [comment]
It should probably be removed completely.

**Comment by mpharrigan** [comment]
Does this not work:
```python
atom_frame.columns = names[:n_cols]
```

http://stackoverflow.com/questions/11346283/renaming-columns-in-pandas

**Comment by jchodera** [comment]
That works too.

**Comment by rmcgibbo** [comment]
@jchodera, I don't think your git command line program is configured with your email address properly -- github isn't tying your github account to your commits.

File: mdtraj/tests/test_mol2.py [hunk 1]
```diff
@@ -109,6 +109,11 @@ def test_mol2_dataframe():
     eq(top.resName[2], "LIG")
     eq(float(top.charge[2]), -0.732600)
 
+def test_mol2_dataframe_status():
+    atoms, bonds = mol2.mol2_to_dataframes(get_fn('adp.mol2'))
+    eq(atoms['charge'][1], 1.3672)
+    eq(atoms['status'][1], '****')
```
Comment (rmcgibbo) [inline mdtraj/tests/test_mol2.py]: silly test failure. just change these to
```
assert atoms['charge'][1] == 1.3672
assert atoms['status'][1]  == '****
```
and it should pass

**Comment by jchodera** [comment]
I think I fixed the failing test case.

**Comment by jchodera** [comment]
> @jchodera, I don't think your git command line program is configured with your email address properly -- github isn't tying your github account to your commits.

Fixed, I hope.

**Comment by jchodera** [comment]
> silly test failure. just change these to

Done.

**Comment by jchodera** [comment]
I don't understand what failed with appveyor...

**Comment by rmcgibbo** [comment]
that's been around for a while.

**Comment by rmcgibbo** [comment]
it's a segfault in pytables on windows :)

**Comment by rmcgibbo** [comment]
https://github.com/mdtraj/mdtraj/pull/913

**Pull request merged**